packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -3636,40 +3614,17 @@ namespace Ifpack2 {
         IFPACK2_BLOCKTRIDICONTAINER_TIMER("BlockTriDi::ComputeResidual::<SeqTag>");
 
         y = y_; b = b_; x = x_;
-        if (is_cuda<execution_space>::value) {
-#if defined(KOKKOS_ENABLE_CUDA)
+        if constexpr (is_device<execution_space>::value) {
```

Comment:
This part probably needs some modifications as the features are not 1-to-1 with the previous code.

The proposed modification calls `ComputeResidualVectorRecommendedCudaVectorSize` if the `is_device<execution_space>::value` is True. In order to be consistent with the previous implementation, we might need a function:

```
template<typename T> 
    static inline int ComputeResidualVectorRecommendedVectorSize(const int blksize,
                                                                     const int team_size) {
    if ( is_cuda<T>::value )
      return ComputeResidualVectorRecommendedCudaVectorSize(blksize, team_size);
    if ( is_hip<T>::value )
      return ComputeResidualVectorRecommendedHIPVectorSize(blksize, team_size);
    if ( is_sycl<T>::value )
      return ComputeResidualVectorRecommendedSYCLVectorSize(blksize, team_size);
    return -1;
    }
```

This comment holds for the other functions such as `ExtractAndFactorizeRecommendedCudaTeamSize`.

Reply:
Ah, my bad I missed that the actually call different functions. I followed your suggestion now.

Reply:
`ExtractAndFactorizeRecommended*TeamSize` is actually only used in functions that are specific to be respective backend anyway.